File: graphs/Questions/Social_Network/C++/Social_Network.cpp
//Question Link:  https://www.codechef.com/problems/SNET

#include<iostream>
#include<algorithm>
#include<vector>
#include<map>

using namespace std;
#define inf -10000000000

int i; 
map< string , vector<string> > adj;
map< string, bool > visited;
long long ans=0;

void dfs(string u)
{
    visited[u]=true;
    ans+=1;
    for(auto it:adj[u])
    {
        if(!visited[it])
        {
            dfs(it);
        }
    }
}

int main() 
{ 
    string s;
    cin>>s;
    long long int n;
    cin>>n;

    for(i=1;i<=n;i++)
    {
        string u,v;
        cin>>u>>v;
        adj[u].__emplace_back(v); 
        adj[v].__emplace_back(u);
    }

    sort(adj[s].begin(),adj[s].end());
    visited[s] = true;
    long long int max = inf;
    string ch;

    for(auto it:adj[s])
    {
        if(!visited[it])
        {
            dfs(it);
            if(ans > max)
            {
                max=ans;
                ch=it;
            }
        }
        ans=0;
    }
    cout<<ch<<"\n"<<max<<"\n";
    return 0; 
}

File: graphs/Questions/Social_Network/Python/social_network.py
## Question Link:  https://www.codechef.com/problems/SNET

from collections import defaultdict

adj = defaultdict(list)
visited = defaultdict(bool)

ans = 0

def dfs(u):
    stack = []
    stack.append(u)
    count = 0

    while len(stack):
        s = stack.pop()

        if not visited[s]:
            count += 1
            visited[s] = True

        for i in adj[s]:
            if not visited[i]:
                stack.append(i)

    return count
        

    
s = input().strip()

n = int(input().strip())

for i in range(n):
    u,v = input().strip().split(" ")
    adj[u].append(v)
    adj[v].append(u)

adj[s].sort()

maxi = 0
ch = "A"

visited[s] = True

for i in adj[s]:
    if not visited[i]:
        ans = dfs(i)
        if ans > maxi:
            maxi = ans
            ch = i


print(ch)
print(maxi)
